### PR TITLE
Feature/document evaluations UI

### PR DESCRIPTION
**Pull request by geclos**

### CODE REVIEW

File: packages/core/drizzle/meta/0044_snapshot.json
Comment (andresgutgon) [inline packages/core/drizzle/meta/0044_snapshot.json:1]: ⚠️ This is mine I think

**Review by andresgutgon — CHANGES_REQUESTED** [review]
Double check you did right the rebase

File: apps/web/src/actions/documents/create.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ export const createDocumentVersionAction = withProject
   .handler(async ({ input, ctx }) => {
     const commitsScope = new CommitsRepository(ctx.project.workspaceId)
     const commit = await commitsScope
-      .getCommitByUuid({ uuid: input.commitUuid, project: ctx.project })
+      .getCommitByUuid({ uuid: input.commitUuid, projectId: ctx.project.id })
```
Comment (geclos) [inline apps/web/src/actions/documents/create.ts:21]: less roundtrips to the db

File: apps/web/src/actions/evaluations/connect.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    const evaluationTemplates = await findAllEvaluationTemplates().then((r) =>
+      r.unwrap(),
+    )
+    const scope = new EvaluationsRepository(ctx.workspace.id)
+    const evaluations = await scope.findAll().then((r) => r.unwrap())
+
+    const selectedTemplates = evaluationTemplates.filter((template) =>
+      input.templateIds.includes(template.id),
+    )
+    const selectedEvaluations = evaluations.filter((evaluation) =>
```
Comment (andresgutgon) [inline apps/web/src/actions/evaluations/connect.ts:33]: This could be SQL

File: apps/web/src/actions/evaluations/connect.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    }),
+  )
+  .handler(async ({ ctx, input }) => {
+    const evaluationTemplates = await findAllEvaluationTemplates().then((r) =>
+      r.unwrap(),
+    )
+    const scope = new EvaluationsRepository(ctx.workspace.id)
+    const evaluations = await scope.findAll().then((r) => r.unwrap())
+
+    const selectedTemplates = evaluationTemplates.filter((template) =>
```
Comment (andresgutgon) [inline apps/web/src/actions/evaluations/connect.ts:30]: This could also be SQL

File: apps/web/src/app/(private)/_data-access/index.ts [hunk 4]
```diff
@@ -170,3 +177,12 @@ export const getProviderLogCached = cache(async (uuid: string) => {
   const scope = new ProviderLogsRepository(workspace.id)
   return await scope.findByUuid(uuid).then((r) => r.unwrap())
 })
+
+export const getEvaluationsByDocumentUuidCached = cache(
```
Comment (andresgutgon) [inline apps/web/src/app/(private)/_data-access/index.ts:181]: Does need to be cached? Is called more than once in a request?

**Review by csansoon — DISMISSED** [review]

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/connect/_components/ConnectToEvaluationStep/EvaluationEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+          </Text.H5>
+        </div>
+      </div>
+    )
+  }
+
+  if (items.length === 1) {
+    return (
+      <div className={className}>
+        <textarea
```
Comment (andresgutgon) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/connect/_components/ConnectToEvaluationStep/EvaluationEditor.tsx:83]: We have `<Textarea />` atom

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/connect/page.tsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+function description(step: number) {
+  switch (step) {
+    case 1:
+      return 'Batch evaluations allow you to analyze a specific amount of logs generated from a dataset.'
+    // TODO: Probably description in step 2 is different from step 1
+    case 2:
+      return 'Batch evaluations allow you to analyze a specific amount of logs generated from a dataset.'
+  }
```
Comment (andresgutgon) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/connect/page.tsx:32]: Same copy in both cases

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/layout.tsx [hunk 7]
```diff
@@ -19,32 +15,30 @@ export default async function DocumentPage({
...
-      commit,
+      projectId,
+      commitUuid,
+      documentUuid,
     })
 
     return (
       <DocumentsLayout
         projectId={projectId}
-        commitUuid={commintUuid}
```
Comment (andresgutgon) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/layout.tsx:39]: omg, it was me?

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/layout.tsx [hunk 8]
```diff
@@ -19,32 +15,30 @@ export default async function DocumentPage({
...
+        commitUuid={commitUuid}
         document={document}
       >
         <DocumentTabs params={params}>{children}</DocumentTabs>
       </DocumentsLayout>
     )
   } catch (error) {
-    // TODO: Show a 404 page within the documents layout, while still showing the sidebar and stuff
-    // For now, we just redirect to documents root if document is not found instead for a cleaner UX
+    if (env.NODE_ENV === 'development') {
```
Comment (andresgutgon) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/layout.tsx:38]: DOn't remove this TODO unless you fix it please

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/connect/_components/ConnectToEvaluationStep/EvaluationEditor.tsx (hunk 5 above)
Comment (geclos) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/connect/_components/ConnectToEvaluationStep/EvaluationEditor.tsx:83]: It has too many baked in styles

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/layout.tsx (hunk 7 above)
Comment (geclos) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/layout.tsx:39]: sip

File: apps/web/src/stores/evaluationTemplates.ts [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+export default function useEvaluationTemplates(
+  opts: SWRConfiguration & { params?: Record<string, unknown> } = {},
+) {
+  const { toast } = useToast()
+
+  const { data = [], ...rest } = useSWR<EvaluationTemplateWithCategory[]>(
+    compact([
+      'evaluationTemplates',
+      ...flatten(Object.entries(opts?.params ?? {})),
```
[the remark this answers is not in the document]
Comment (geclos) [reply apps/web/src/stores/evaluationTemplates.ts:18]: could be anything honestly just copied from evaluations store

File: apps/web/src/app/(private)/_data-access/index.ts (hunk 4 above)
Comment (geclos) [reply apps/web/src/app/(private)/_data-access/index.ts:181]: 🤷🏼 doesn't hurt no?

**Review by andresgutgon — DISMISSED** [review]

**Pull request merged**